fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/PartitionAvailability.java

```diff
@@ -0,0 +1,86 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.table.availability;
+
+import com.yahoo.bard.webservice.data.config.names.TableName;
+import com.yahoo.bard.webservice.table.Column;
+import com.yahoo.bard.webservice.table.resolver.DataSourceConstraint;
+import com.yahoo.bard.webservice.util.IntervalUtils;
+import com.yahoo.bard.webservice.util.SimplifiedIntervalList;
+
+import org.joda.time.Interval;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import javax.validation.constraints.NotNull;
+
+/**
+ * An implementation of Availability that unions same columns from Druid tables together.
+ */
+public class PartitionAvailability implements Availability {
+
+    private final Set<Availability> sourceAvailabilities;
+    private final Set<Column> columns;
+    private final Function<DataSourceConstraint, Set<Availability>> partitionFunction;
+
+    /**
+     * Constructor.
+     *
+     * @param sourceAvailabilities  The set of availabilities that have the same columns
+     * @param columns  The set of configured columns
+     * @param partitionFunction  A function that transform a DataSourceConstraint to a set of
+     * Availabilities
```

Comment:
More detail needed on what the expectations are of the partition function. (ie. what is it partitioning on, how is it partitioning, etc.)